src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -459,8 +459,10 @@ public void startActionListener(int timeout) {
      */
     public static void run(Extension extension) throws IOException {
         logger.info("Starting extension " + extension.getExtensionSettings().getExtensionName());
-        @SuppressWarnings("unused")
         ExtensionsRunner runner = new ExtensionsRunner(extension);
+        // initialize the transport service
+        NettyTransport nettyTransport = new NettyTransport(runner);
+        nettyTransport.initializeExtensionTransportService(runner.getSettings(), new ThreadPool(runner.getSettings()), runner);
```

Comment:
I'm OK to approve the PR like this, but creating a thread pool that lives inside the TransportService seems not correct.  I think this will be resolved in our discussion as part of #183.